addons/account/models/account_move_line.py

```diff
@@ -243,18 +243,32 @@ class AccountMoveLine(models.Model):
     )
 
     # === Reconciliation fields === #
+    # Technical field used by the search filter to include partial reconciliations up to a date.
+    reconcile_until = fields.Date(
+        store=False,
+        search='_search_reconcile_until',
+    )
     amount_residual = fields.Monetary(
         string='Residual Amount',
         compute='_compute_amount_residual', store=True,
         currency_field='company_currency_id',
         help="The residual amount on a journal item expressed in the company currency.",
     )
+    amount_residual_at_date = fields.Monetary(
+        compute='_compute_amount_residual_at_date',
+        currency_field='company_currency_id',
+        help="The residual amount at a specific date expressed in the company currency."
+    )
```

Comment:
actually, the more I think of it and the more I think we shouldn't add new fields for that